Resolve spurious linkchecker timeouts

mattjala opened the pull request:
- Remove unneeded packages from linkchecker installation

The texlive packages constituted over a gigabyte of data to install, but seem to have no impact on the linkchecker workflow. Removing this drops the installation time from ~9 minutes to ~30 seconds. 

- Reduce number of linkchecker threads 
 
The [current linkchecker](https://github.com/linkchecker/linkchecker) often spuriously fails on valid links due to timeouts (see [here](https://github.com/HDFGroup/hdf5/actions/runs/15586347547/job/43893439400), [here](https://github.com/HDFGroup/hdf5/actions/runs/15637693352/job/44057051203), and [here](https://github.com/HDFGroup/hdf5/actions/runs/15353948295/job/43208633948) as examples).

This seems to be due to the target websites rejecting the connection, potentially due to the linkchecker using multiple threads to send many requests to the same domain.

This PR reduces the number of threads the linkchecker uses from the default (10) to 2. This does not substantially impact the time the linkchecker takes to run.

Resolves #5439

jhendersonHDF commented:
Have we verified that this actually resolves #5439 in a meaningful way? Usually, increasing a timeout value is just asking for this to continue happening in the future.

brtnfld commented:
At least for PR #5506, the addition of the timeout, along with `--no-robots`, did not provide any improvement.

mattjala commented:
> At least for PR #5506, the addition of the timeout, along with `--no-robots`, did not provide any improvement.

For that PR, it looks like the linkchecker is failing due to real problems with existing links - `https://www.mcs.anl.gov/~thakur/papers/romio-coll.ps.gz` and `http://www.mcs.anl.gov/~thakur/papers/mpio-high-perf.ps.gz` are 404 pages now. So it makes sense that altering the timeout didn't change anything

mattjala commented:
> Have we verified that this actually resolves #5439 in a meaningful way? Usually, increasing a timeout value is just asking for this to continue happening in the future.

The CI failures are sporadic, so I'm not sure how we can test it directly beyond making the change and seeing if the frequency of failures change. The timeout errors happen when trying to connect to websites that are, to all appearances, online and operational, so increasing the timeout value should have an effect.

brtnfld commented:
https://web.cels.anl.gov/~thakur/papers/romio-coll.pdf
https://web.cels.anl.gov/~thakur/papers/users-guide.pdf

mattjala commented:
> https://web.cels.anl.gov/~thakur/papers/romio-coll.pdf https://web.cels.anl.gov/~thakur/papers/users-guide.pdf

Links are updated in #5605

mattjala committed:
Remove unnecessary workflow installs

mattjala committed:
Remove extra font installation

mattjala committed:
Remove texlive install

mattjala committed:
Reduce the number of linkchecker threads

mattjala committed:
Reduce linkchecker threads further

mattjala commented:
Changed the approach to reducing the number of threads instead of increasing timeout. Also removed unneeded package installs that seriously dropped the time needed for the install step (9 min -> 30 sec)

mattjala committed:
Add tuxfamily to linkchecker ignore list

mattjala commented:
Added [tuxfamily](https://eigen.tuxfamily.org/index.php?title=Main_Page) to the list of ignored URLs since the site itself seems unreliable and is [causing linkchecker failures](https://github.com/mattjala/hdf5/actions/runs/15714785349/job/44281698403).

We could also simply remove the two links to that domain - they're both in About.dox, and are slightly redundant since we also link to the eigen Gitlab repo, which itself links to that domain.
